docs/index.rst

```diff
@@ -37,7 +37,7 @@ processing, and data analysis.
 
    index_getting_started
    index_user_docs
-   index_dev
+   development/index
```

Comment:
I think there was a reason why we made the index at top level (`index_dev`) vs `development/index` but I don't remember why now. @saimn , do you remember?

Reply:
Just going to xref https://github.com/astropy/astropy/pull/14867 here in case we need to revert this particular change in the future.